learn/work/sparse-recipe/index.qmd

```diff
@@ -0,0 +1,254 @@
+
+The second change is that a number of steps are now able to take sparse vectors as input while preserving sparsity. You can see a full [list of these steps](../../../find/sparse/index.qmd) at the link.
+
+Most of the changes with regard to sparsity are done to minimize the changes the user needs to make to their code. This means that in many cases you don't need to change anything, the steps will know when to produce sparse data or not.
+
+When a recipe is used in a workflow and it is being `fit()`, an internal check is being done to figure out whether or not to produce sparse features. This check looks at the sparsity of the data itself, what model is being used, and the recipe. Since [only some models](../../../find/sparse/index.qmd) support sparsity this is the first check. 
+
+A rough estimate of the sparsity of the data that will come out of the recipe is calculated. This is done using the input data set, and the steps present in the recipe. But since this check has to happen before the recipe is prepped, it will be quite simple. What this means in practice is that it is fairly good at estimating the sparsity that is produced by sparsity-generating steps, but it isn't able to detect if those variables are passed to a different step that doesn't preserve it.
+
+The following recipe would be accurately estimated as the dummies produced by `step_dummy()` aren't passed to any other steps.
+
+```{r}
+#| eval: false
+recipe(outcome ~ ., data = data_train) |>
+  step_normalize(all_numeric_predictors()) |>
+  step_dummy(all_nominal_predictors())
+```
+
+But the next recipe would have the same sparsity estimate despite not being able to produce any sparsity since `step_normalize()` can't preserve the sparsity as it subtracts a constant value.
+
+```{r}
+#| eval: false
+recipe(outcome ~ ., data = data_train) |>
+  step_dummy(all_nominal_predictors()) |>
+  step_normalize(all_numeric_predictors())
+```
+
+If you were able to modify the above recipe to use `step_scale()` instead of `step_normalize()` then the estimate is still valid as `step_scale()` is a sparsity-preserving step.
+
+```{r}
+#| eval: false
+recipe(outcome ~ ., data = data_train) |>
+  step_dummy(all_nominal_predictors()) |>
+  step_scale(all_numeric_predictors())
+```
+
+Is it for this reason the steps that produce sparsity have the `sparse` argument. It defaults to `"auto"`, which means that the estimating process in workflows decides whether or not sparsity should be created. This argument can take two other values `"yes"` and `"no"`. If you know for certain that the recipe should or shouldn't produce sparsity you can overwrite with this argument.
+
+This means that the below recipe is good again as it wouldn't try to produce sparse vectors that will immediately be turned into dense vectors again.
```

Comment:
```suggestion
This means the recipe below wouldn't try to initially produce sparse vectors since they will immediately be turned into dense vectors by the next step.
```